src/persistence/settings.cpp

```diff
@@ -1379,6 +1409,48 @@ void Settings::setCircleExpanded(int id, bool expanded)
     circleLst[id].expanded = expanded;
 }
 
+void Settings::addFriendRequest(const QString &friendAddress, const QString &message)
+{
+    QMutexLocker locker{&bigLock};
+    QPair<QString, QString> request(friendAddress, message);
+
+    if (friendRequests.indexOf(request) != -1)
+        return;
+
+    friendRequests.push_back(request);
+    ++unreadFriendRequests;
+}
+
+unsigned int Settings::getUnreadFriendRequests() const
+{
+    QMutexLocker locker{&bigLock};
```

Comment:
No, the aforementioned classes offer reference counting, which is not what the settings class needs. A possible alternative is to split up the settings into multiple classes, which the main settings class would access when it needs to, and using RAII, accessing theses classes could be  mutex protected. In general, I think qTox needs more looser coupling.